src/main/java/com/github/reugn/devtools/services/RestService.java

```diff
@@ -1,54 +1,103 @@
+
+import javafx.application.Platform;
+
 public class RestService {
 
+	private static final ExecutorService EXECUTOR = Executors.newSingleThreadExecutor();
+	public static final List<Request> REQ_HISTORY_LIST = new ArrayList<>();
+	private static RestAPIController CONTROLLER;
+	
     private RestService() {
     }
 
     private static final int defaultConnectTimeout = 60000;
 
-    public static RestResponse request(String requestMethod, String uri, Map<String, String> headers,
-                                       String body) throws Exception {
-        HttpURLConnection conn = (HttpURLConnection) buildURL(uri).openConnection();
-        conn.setRequestMethod(requestMethod);
-        for (Map.Entry<String, String> h : headers.entrySet()) {
-            conn.setRequestProperty(h.getKey(), h.getValue());
-        }
-        conn.setConnectTimeout(defaultConnectTimeout);
-        conn.setInstanceFollowRedirects(false);
-
-        String responseBody = "";
-        int status;
-        long t1 = System.currentTimeMillis();
-        if (!body.isEmpty()) {
-            conn.setDoOutput(true);
-            setRequestBody(conn.getOutputStream(), body);
-        }
-        try {
-            status = conn.getResponseCode();
-            responseBody = getResponseBody(conn.getInputStream());
-        } catch (IOException e) {
-            status = conn.getResponseCode();
-            responseBody = getResponseBody(conn.getErrorStream());
-        }
-        long t2 = System.currentTimeMillis() - t1;
+    public static void addToHistoryReqList(List<Request> reqs) {
+    	REQ_HISTORY_LIST.addAll(reqs);
+    }
+    
+	private static RestResponse request(String requestMethod, String uri, Map<String, String> headers, String body) throws Exception {
+		HttpURLConnection conn = (HttpURLConnection) buildURL(uri).openConnection();
+		conn.setRequestMethod(requestMethod);
+		for (Map.Entry<String, String> h : headers.entrySet()) {
+			conn.setRequestProperty(h.getKey(), h.getValue());
+		}
+		conn.setConnectTimeout(defaultConnectTimeout);
+		conn.setInstanceFollowRedirects(false);
+
+		String responseBody = "";
+		int status;
+		long t1 = System.currentTimeMillis();
+		if (!body.isEmpty()) {
+			conn.setDoOutput(true);
+			setRequestBody(conn.getOutputStream(), body);
+		}
+		try {
+			status = conn.getResponseCode();
+			responseBody = getResponseBody(conn.getInputStream());
+		} catch (IOException e) {
+			status = conn.getResponseCode();
+			responseBody = getResponseBody(conn.getErrorStream());
+		}
+		long t2 = System.currentTimeMillis() - t1;
 
-        String responseHeaders = getResponseHeaders(conn.getHeaderFields());
-        conn.disconnect();
-        return new RestResponse(status, responseBody, responseHeaders, t2);
+		String responseHeaders = getResponseHeaders(conn.getHeaderFields());
+		conn.disconnect();
+
+		Request req = new Request(uri, requestMethod, headers, body);
+		REQ_HISTORY_LIST.add(req);
```

Comment:
Don't add, if the exact request already exists in the list and move it to the bottom as a last used.

Reply:
Done